kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -272,7 +272,7 @@ object Async:
             def toFuture(using E <:< Throwable, Frame): Future[A] < IO =
                 IO {
                     val r = scala.concurrent.Promise[A]()
-                    self.onComplete { v =>
```

Comment:
`toFuture`, `map`, `flatMap`, and `mapResult` were all incorrect. The compiler was resolving the `onComplete` extension method for `Fiber` and not the correct `IOPromise.onComplete`, which produced an `Unit < IO` and didn't execute anything. The rename to `onResult` resolves the ambiguity.